napari/components/layerlist.py

```diff
@@ -129,6 +129,9 @@ def _update_name(self, event):
     def insert(self, index: int, value: Layer):
         """Insert ``value`` before index."""
```

Comment:
Could you please add the proper annotation to the docstring? Maybe also to class docstring? 

Behavior from PR is obvious when you think longer, but maybe not obvious for a user who enter to `napari` world and sees the container that name contains "List"

Reply:
Not sure what you're referring to here. You mean replacing ``value`` with ``Layer`` in the docstring of `insert`?

Reply:
as you raise an exception, not a silent movement, there is no need for additional docstring.